src/main/java/domen/rideapp/infrastructure/GoogleMapsConfiguration.java

```diff
@@ -28,12 +29,18 @@ GoogleMapsUrlBuilder googleMapsUrlBuilder(GoogleMapsConfig config) {
     }
 
     @Bean
-    MapService googleMapsServiceProd(GoogleMapsClient client,
-                                     ConversionService conversionService,
-                                     GoogleMapsConfig config) {
+    MapService googleMapsService(FeatureFlagsConfig featureFlagsConfig,
+                                 GoogleMapsClient client,
+                                 ConversionService conversionService,
+                                 GoogleMapsConfig config) {
+        if (!featureFlagsConfig.googleMapsEnabled()) {
```

Comment:
Zamiast ifa + klasy FeatureFlagsConfig to stwórz 2 beany MapService w zalezności od ConditionalOnProperty ([spring-feature-flags](https://www.baeldung.com/spring-feature-flags))